source/tests/consistent/model/test_dos.py

```diff
@@ -0,0 +1,190 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+import unittest
+from typing import (
+    Any,
+)
+
+import numpy as np
+
+from deepmd.dpmodel.model.dos_model import DOSModel as DOSModelDP
+from deepmd.dpmodel.model.model import get_model as get_model_dp
+from deepmd.env import (
+    GLOBAL_NP_FLOAT_PRECISION,
+)
+
+from ..common import (
+    INSTALLED_PT,
+    INSTALLED_TF,
+    CommonTest,
+)
+from .common import (
+    ModelTest,
+)
+
+if INSTALLED_PT:
+    from deepmd.pt.model.model import get_model as get_model_pt
+    from deepmd.pt.model.model.dos_model import DOSModel as DOSModelPT
+else:
+    DOSModelPT = None
+if INSTALLED_TF:
+    from deepmd.tf.model.dos import DOSModel as DOSModelTF
+else:
+    DOSModelTF = None
+from deepmd.utils.argcheck import (
+    model_args,
+)
+
+
+class TestDOS(CommonTest, ModelTest, unittest.TestCase):
+    @property
+    def data(self) -> dict:
+        return {
+            "type_map": ["O", "H"],
+            "descriptor": {
+                "type": "se_e2_a",
+                "sel": [20, 20],
+                "rcut_smth": 1.8,
+                "rcut": 6.0,
+                "neuron": [2, 4, 8],
+                "resnet_dt": False,
+                "axis_neuron": 8,
+                "precision": "float64",
+                "seed": 1,
+            },
+            "fitting_net": {
+                "type": "dos",
+                "numb_dos": 2,
+                "neuron": [4, 4, 4],
+                "resnet_dt": True,
+                "numb_fparam": 0,
+                "precision": "float64",
+                "seed": 1,
+            },
+        }
+
+    tf_class = DOSModelTF
+    dp_class = DOSModelDP
+    pt_class = DOSModelPT
+    args = model_args()
+
+    def get_reference_backend(self):
+        """Get the reference backend.
+
+        We need a reference backend that can reproduce forces.
+        """
+        if not self.skip_pt:
+            return self.RefBackend.PT
+        if not self.skip_tf:
+            return self.RefBackend.TF
+        if not self.skip_dp:
+            return self.RefBackend.DP
+        raise ValueError("No available reference")
+
+    @property
+    def skip_tf(self):
+        return True  # need to fix tf consistency
+
```

Comment:
Not sure why the PT backend is not consistent with the TF backend. I think they are consistent at the fitting net level and the descriptor level. Need further investigation.